WordPress/src/main/java/org/wordpress/android/GCMMessageService.java

```diff
@@ -335,8 +366,55 @@ private void showNotificationForBuilder(NotificationCompat.Builder builder, Cont
     }
 
     // Clear all notifications
-    private void handleBadgeResetPN() {
-        removeAllNotifications(this);
+    private void handleBadgeResetPN(Bundle data) {
+        if (data != null && data.containsKey(PUSH_ARG_NOTE_ID)) {
+            removeNotificationWithNoteIdFromSystemBar(this, data);
+            //now that we cleared the specific notif, we can check and make any visual updates
+            if (sActiveNotificationsMap.size() > 0) {
+                Bitmap largeIconBitmap = null;
+                // here notify the existing group notification by eliminating the line that is now gone
+                String title = StringEscapeUtils.unescapeHtml(data.getString(PUSH_ARG_TITLE));
+                if (title == null) {
+                    title = getString(R.string.app_name);
+                }
+                String message = StringEscapeUtils.unescapeHtml(data.getString(PUSH_ARG_MSG));
+
+                if (sActiveNotificationsMap.size() == 1) {
+                    //only one notification remains, so get the proper message for it and re-instate in the system dashboard
+                    Bundle remainingNote = sActiveNotificationsMap.values().iterator().next();
+                    if (remainingNote != null) {
+                        String remainingNoteTitle = StringEscapeUtils.unescapeHtml(remainingNote.getString(PUSH_ARG_TITLE));
+                        if (!TextUtils.isEmpty(remainingNoteTitle)) {
+                            title = remainingNoteTitle;
+                        }
+                        String remainingNoteMessage = StringEscapeUtils.unescapeHtml(remainingNote.getString(PUSH_ARG_MSG));
+                        if (!TextUtils.isEmpty(remainingNoteMessage)) {
+                            message = remainingNoteMessage;
+                        }
+                        largeIconBitmap = getLargeIconBitmap(remainingNote.getString("icon"),
+                                shouldCircularizeNoteIcon(remainingNote.getString(PUSH_ARG_TYPE)));
+                    }
+                }
+
+                NotificationCompat.Builder builder = getNotificationBuilder(title, message);
+
+                if (largeIconBitmap == null) {
+                    largeIconBitmap = getLargeIconBitmap(data.getString("icon"), shouldCircularizeNoteIcon(PUSH_TYPE_BADGE_RESET));
+                }
+
+                if (largeIconBitmap != null) {
+                    builder.setLargeIcon(largeIconBitmap);
+                }
+
+                showGroupNotificationForBuilder(builder, message);
+            }
+        } else {
+            //TODO: check if this is necessary:
```

Comment:
Can we remove that else block and make this more explicit. Maybe something like this on line 370:

```java
if (data == null || !data.containsKey(PUSH_ARG_NOTE_ID))  {
    // ignore the reset-badge PN if it's a global one
    return;
}
...
```